### PR TITLE
Vagabond/#5578

### LINKED ISSUE #5578: Rename 'useJsonForms' hook to avoid confusion
## The suggestion

There are currently two different `useJsonForms` hooks in use -- one is part of the core JSONForms package and provides full context of the JSON Forms state. The other is one we've created which we use to manage the forms and provide an overall "wrapper" for the entire structure.

This can be mildly confusing -- most individual components use the first one, and the implementations of JSON forms use the latter. It would be fairly trivial to rename our custom hook to something distinct, for example "useJsonFormsHandler" or "useJsonFormsWrapper".

It's not a huge point of confusion, but it's very simple to fix, so probably worth it IMHO.

### Why should we invest time in this?

- Less confusion when trying to understand JSON forms code
- Easier to search for the correct thing

### Are there any risks associated with this change?

Doubt it, the compiler will yell at us if we miss any references to the renamed hook

### How much effort is required?

Less than 1 hour.

### Agreed Solution

... ? (My vote is for `useJsonFormsHandler`)

### PR DESCRIPTION
**Pull request by vagabond-0**
Fixes #5578

# 👩🏻‍💻 What does this PR do?

Rename every the function useJsonForm to useJsonFormsHandler

To avoid confusion

No change in ui
## 💌 Any notes for the reviewer?

Fixes #

# 👩🏻‍💻 What does this PR do?

Nil

NIl

NIl

### CODE REVIEW

File: client/packages/programs/src/JsonForms/useJsonForms.tsx [hunk 1]
```diff
@@ -106,13 +106,13 @@ export type JsonFormData<R> = {
 };
 
 /**
- * This hook add provides functionality to save form data and keep track if form data has been
+  * This hook add provides functionality to save form data and keep track if form data has been
```
Comment (CarlosNZ) [inline client/packages/programs/src/JsonForms/useJsonForms.tsx:109]: Could you remove this whitespace change, please?

File: client/packages/system/src/Patient/PatientView/PatientView.tsx [hunk 2]
```diff
@@ -199,7 +199,7 @@ const PatientDetailView = ({
       };
 
   const { JsonForm, saveData, isSaving, isDirty, validationError } =
-    useJsonForms(
+  useJsonFormsHandler (
```
Comment (CarlosNZ) [inline client/packages/system/src/Patient/PatientView/PatientView.tsx:202]: Revert whitespace change (If you have Prettier set up, this should happen automatically)

File: client/packages/system/src/Patient/ProgramEnrolment/Components/DetailModal.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ export const ProgramDetailModal: FC = () => {
     handleSave
   );
   const { JsonForm, isLoading, isSaving, saveData, isDirty, validationError } =
-    useJsonForms(
+  useJsonFormsHandler(
```
Comment (CarlosNZ) [inline client/packages/system/src/Patient/ProgramEnrolment/Components/DetailModal.tsx:64]: Same here.

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Thanks for contributing @vagabond-0

Aside from those minor whitespace changes, could you also rename the *file* to `useJsonFormsHandler.tsx` as well please?

Thanks

**Comment by vagabond-0** [comment]
Removed white spaces and changed the file name to useJsonFormsHandler.tsx

**Review by CarlosNZ — APPROVED** [review]
LGTM!

Thanks for contributing.

**Pull request merged**